test/Integration/SqlTriggerBindingIntegrationTests.cs

```diff
@@ -872,5 +872,68 @@ public async Task NewUserFunctionId_Migration_Test()
             Assert.True(1 == (int)this.ExecuteScalar($@"SELECT 1 FROM {GlobalStateTableName} WHERE UserFunctionID = N'{userFunctionId}'"), $"{GlobalStateTableName} should have {userFunctionId} row on successful migration");
             Assert.True(lastSyncVersion == (long)this.ExecuteScalar($@"SELECT LastSyncVersion FROM {GlobalStateTableName} WHERE UserFunctionID = N'{userFunctionId}'"), $"{GlobalStateTableName} should have {userFunctionId} row woth on successful migration");
         }
+
+        /// <summary>
+        /// Ensures that the user function gets invoked for each of the insert, update and delete operation after migration seamlessly.
+        /// </summary>
+        [RetryTheory]
+        [SqlInlineData()]
+        [UnsupportedLanguages(SupportedLanguages.Java)] // test timing out for Java
```

Comment:
Have you verified manually that this works for Java? If the test is failing because of a timeout that's a bit worrisome to me

Reply:
tested and it creates separate leases table.